Shared/Samples/Create and save map/CreateAndSaveMapView.swift

```diff
@@ -0,0 +1,419 @@
+    enum OperationalDataOption: CaseIterable {
+        /// No operational data.
+        case none
+        /// Operational data for time zones.
+        case timeZones
+        /// U.S. census tracts.
+        case census
+        
+        /// The url of the layer.
+        private var url: URL? {
+            switch self {
+            case .none:
+                nil
+            case .timeZones:
+                URL(string: "https://sampleserver6.arcgisonline.com/arcgis/rest/services/WorldTimeZones/MapServer")!
+            case .census:
+                URL(string: "https://sampleserver6.arcgisonline.com/arcgis/rest/services/Census/MapServer")!
+            }
+        }
+        
+        /// The map image layer that corresponds to the option.
+        var layer: ArcGISMapImageLayer? {
+            url.map(ArcGISMapImageLayer.init(url:))
+        }
+        
+        /// The label for user interface purposes.
+        var label: String {
+            switch self {
+            case .none:
+                "None"
+            case .timeZones:
+                "Time Zone"
+            case .census:
+                "Census"
+            }
+        }
+    }
+}
+
+private extension CreateAndSaveMapView {
+    /// The status of the create and save workflow.
+    enum Status: Equatable {
+        /// The portal is loading.
+        case loadingPortal
+        /// The portal failed to load.
+        case failedToLoadPortal
+        /// The map is being created.
+        case creatingMap
+        /// The map is being saved to the portal.
+        case savingMapToPortal
+        /// The map failed to save to the portal.
+        case failedToSaveMap
+        /// The map was saved successfully to the portal.
+        case mapSavedSuccessfully(Map)
+        /// The map is being deleted from the portal.
+        case deletingMap
+        /// The map was successfully deleted from the portal.
+        case deletedSuccessfully
+        /// The map failed to delete from the portal.
+        case failedToDelete
+    }
+}
+
+private extension CreateAndSaveMapView {
+    /// Sets up the authenticator to handle challenges.
+    func setupAuthenticator() {
+        // Setting the challenge handlers here when the model is created so user is prompted to enter
+        // credentials every time trying the sample. In real world applications, set challenge
+        // handlers at the start of the application.
+        
+        // Sets authenticator as ArcGIS and Network challenge handlers to handle authentication
+        // challenges.
+        ArcGISEnvironment.authenticationManager.handleChallenges(using: authenticator)
+        
+        // In your application you may want to uncomment this code to persist
+        // credentials in the keychain.
+        // setupPersistentCredentialStorage()
+    }
+    
+    /// Stops the authenticator from handling the challenges and clears credentials.
+    nonisolated func teardownAuthenticator() async {
```

Comment:
```suggestion
    func teardownAuthenticator() async {
```

Is `nonisolated` needed here?